alpine/workfile.py

```diff
@@ -494,11 +498,18 @@ def wait_until_finished(self, process_id, verbose=False, query_time=10, timeout=
                         " It now has status '{2}'.".format(process_id, timeout, stop_status))
 
                 if verbose:
-                    print("\rWorkflow in progress for ~{0:.2f} seconds.".format(wait_total)),
+                    print("\rWorkflow in progress for ~{0:.1f} seconds.".format(wait_total)),
 
                 time.sleep(query_time)
 
-                if workflow_status == "FAILED":
-                    raise RunFlowFailureException("The workflow with process ID: <{0}> failed.".format(process_id))
                 workflow_status = self.query_status(process_id)
-            return workflow_status
+
+            if verbose:
+                print("")
+
+            # If workflow is finished - find the final status, otherwise return status in progress.
+            time.sleep(1)
```

Comment:
I would suggest to add a try except block for sleep and get the status from results.